src/redux/stores/nftDetail/nftDetailHelpers.ts

```diff
@@ -0,0 +1,68 @@
+import { Web3Provider } from '@ethersproject/providers';
+import erc721AbiContract from '@openzeppelin/contracts/build/contracts/ERC721.json';
+import erc1155Contract from '@openzeppelin/contracts/build/contracts/ERC1155.json';
+import { BigNumber, ethers } from 'ethers';
+import { defaultAbiCoder } from 'ethers/lib/utils';
+
+import { NftTransactionLog } from '../../../entities/NftTransactionLog/NftTransactionLog';
+import {
+  transformTransactionReceiptToNftTransactionLog,
+} from '../../../entities/NftTransactionLog/NftTransactionLogTransformers';
+import { getUniqueArrayChildren } from '../../../helpers/array';
+
+export const getErc721Logs = async (
+  chainId: number,
+  collectionToken: string,
+  provider: Web3Provider,
+  tokenId: number,
+): Promise<NftTransactionLog[]> => {
+  const contract = new ethers.Contract(collectionToken, erc721AbiContract.abi, provider);
+  const transferFilter = contract.filters.Transfer(null, null, tokenId);
+
+  const events = await contract.queryFilter(transferFilter, 0);
+
+  const transactionReceipts = await Promise.all(events.map(event => event.getTransactionReceipt()));
+  const transactionReceiptBlocks = await Promise.all(transactionReceipts.map(receipt => provider.getBlock(receipt.blockNumber)));
+
+  const logs = transactionReceipts.map((transactionReceipt, index) => transformTransactionReceiptToNftTransactionLog(
+    transactionReceipt,
+    transactionReceiptBlocks[index].timestamp,
+    transactionReceipt.from,
+  )).sort((a, b) => b.timestamp - a.timestamp);
+
+  return getUniqueArrayChildren(logs, 'transactionHash');
+};
+
+export const getErc1155Logs = async (
+  chainId: number,
+  collectionToken: string,
+  provider: Web3Provider,
+  tokenId: number,
+): Promise<NftTransactionLog[]> => {
+  const contract = new ethers.Contract(collectionToken, erc1155Contract.abi, provider);
+  const transferFilter = contract.filters.TransferSingle(null, null, null, null);
```

Comment:
@gpxl-dev Any idea why `contract.filters.TransferSingle(null, null, null, tokenId);` doesn't work?

```
Error: cannot filter non-indexed parameters; must be null (argument="contract.id", value="10055", code=INVALID_ARGUMENT, version=abi/5.7.0)
```

https://docs.openzeppelin.com/contracts/3.x/api/token/erc1155#IERC1155-TransferSingle-address-address-address-uint256-uint256-

Reply:
`event TransferSingle(address indexed _operator, address indexed _from, address indexed _to, uint256 _id, uint256 _value);`

Yes - as the error suggests the parameter you are trying to filter by (`_id` in the signature above) is not indexed. You can only filer by indexed params.

So in this case you would need to get all of the transfer events and manually filter them. It sucks, I know...